homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -20,54 +21,71 @@
 from .const import (
     CONF_VOLUMES,
     DEFAULT_DSM_VERSION,
-    DEFAULT_NAME,
     DEFAULT_PORT,
     DEFAULT_PORT_SSL,
     DEFAULT_SSL,
 )
 from .const import DOMAIN  # pylint: disable=unused-import
 
+_LOGGER = logging.getLogger(__name__)
+
+
+def _discovery_schema_with_defaults(discovery_info=None):
+    return vol.Schema(_ordered_shared_schema(discovery_info))
+
+
+def _user_schema_with_defaults(user_input=None):
+    user_schema = {
+        vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+    }
+    user_schema.update(_ordered_shared_schema(user_input))
+
+    return vol.Schema(user_schema)
+
+
+def _ordered_shared_schema(schema_input=None):
```

Comment:
Can you remove `param=None` ?

Doing a get on a None will result to `AttributeError: 'NoneType' object has no attribute 'get'`.

And `discovery_info` and `user_input` inputs are not None on the entry

Reply:
removed from the schema builder functions.